lib/pages/add_bridge/add_bridge.dart

```
@@ -176,6 +190,118 @@ class BotController extends State<AddBridge> {
     Logs().i('No message received from bot within the wait time');
   }
 
+  Future<http.Response> requestGet(String url) async {
+    return await http.get(Uri.parse(url), headers: headers);
+  }
+
+  Future<http.Response> requestPost(String url, {dynamic body}) async {
+    return await http.post(Uri.parse(url), headers: headers, body: body);
+  }
+
+  Future<void> pingBridgeAPI(SocialNetwork network) async {
+    final bridgePathIdentifier  = getBridgePath(network);
+    final userId = client.userID;
+
+    final url = 'https://matrix.$serverUrl/_matrix/$bridgePathIdentifier/provision/v3/whoami?user_id=$userId';
+
+    final response = await requestGet(url);
+
+    interpretBridgeResponse(response);
+
+    if (response.statusCode == 200) {
+      if (kDebugMode) {
+        print('Successful ping');
+      }
+    } else {
+      if (kDebugMode) {
+        print('Error: ${response.statusCode}');
```

This error is not propagated or shown to user